README.md
# [Material Kit](https://demos.creative-tim.com/material-kit/index.html) [![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&logo=twitter)](https://twitter.com/home?status=Material%20Kit%20is%20a%20Free%20Bootstrap%204%20UI%20Kit%20%E2%9D%A4%EF%B8%8F%0Ahttps%3A//demos.creative-tim.com/material-kit/index.html%20%23bootstrap%20%23material%20%23design%20%23uikit%20%23freebie%20%20via%20%40CreativeTim)


 ![version](https://img.shields.io/badge/version-2.0.5-blue.svg)  ![license](https://img.shields.io/badge/license-MIT-blue.svg) [![GitHub issues open](https://img.shields.io/github/issues/creativetimofficial/material-kit.svg?maxAge=2592000)](https://github.com/creativetimofficial/material-kit/issues?q=is%3Aopen+is%3Aissue) [![GitHub issues closed](https://img.shields.io/github/issues-closed-raw/creativetimofficial/material-kit.svg?maxAge=2592000)](https://github.com/creativetimofficial/material-kit/issues?q=is%3Aissue+is%3Aclosed) [![Join the chat at https://gitter.im/NIT-dgp/General](https://badges.gitter.im/NIT-dgp/General.svg)](https://gitter.im/creative-tim-general/Lobby) [![Chat](https://img.shields.io/badge/chat-on%20discord-7289da.svg)](https://discord.gg/E4aHAQy)


![Product Gif](https://s3.amazonaws.com/creativetim_bucket/github/gif/material-kit-free.gif)

Material Kit is a Free Bootstrap 4 UI Kit with a fresh, new design inspired by Google's material design. You asked for it, so we built it. It's a great pleasure to introduce to you the material concepts in an easy to use and beautiful set of components. Along with the restyling of the Bootstrap elements, you will find three fully-coded example pages, to help you design your next project.

Material Kit makes use of light, surface and movement. It uses a deliberate color choice, edge-to-edge imagery and large scale typography. The general layout resembles sheets of paper following multiple different layers, so that the depth and order is obvious. The navigation stays mainly on the left and the actions on the right.

This new design has elements that have been the result of research regarding ink and paper and the way objects and materials interact in real life. The result is a beautiful and consistent set of elements that can get you started with your next project. Material Kit is a great tool if you are looking to create a web presence for your Android application and need to be consistent, leaving the impression of visually similar elements. It is also a great resource in its own right, looking gorgeous and helping you build your web pages.

Material Kit is based on the Github Repo from Fezvrasta who did an amazing job creating the backbone for the material effects, animations and transitions. Big thanks to his team for their forward thinking efforts.


## Table of Contents

* [Versions](#versions) 
* [Demo](#demo)
* [Quick Start](#quick-start)
* [Documentation](#documentation)
* [File Structure](#file-structure)
* [Browser Support](#browser-support)
* [Resources](#resources)
* [Reporting Issues](#reporting-issues)
* [Technical Support or Questions](#technical-support-or-questions)
* [Licensing](#licensing)
* [Useful Links](#useful-links)



## Versions

[<img src="https://s3.amazonaws.com/creativetim_bucket/github/html.png" width="60" height="60" />](https://www.creative-tim.com/product/material-kit)[<img src="https://s3.amazonaws.com/creativetim_bucket/github/react.svg" width="60" height="60" />](https://www.creative-tim.com/product/material-kit-react)


| HTML | React  |
| --- | ---  |
| [![Material Kit  HTML](https://s3.amazonaws.com/creativetim_bucket/products/38/thumb/opt_mk_thumbnail.jpg)](https://www.creative-tim.com/product/material-kit)  | [![Material Kit  React](https://s3.amazonaws.com/creativetim_bucket/products/83/thumb/opt_mk_react_thumbnail.jpg)](https://www.creative-tim.com/product/material-kit-react)   



## Demo

- [Index Page](https://demos.creative-tim.com/material-kit/index.html)
- [Buttons Section](https://demos.creative-tim.com/material-kit/index.html#buttons)
- [Navigation Section](https://demos.creative-tim.com/material-kit/index.html#navigation)
- [Notifications Section](https://demos.creative-tim.com/material-kit/index.html#notifications)
- [User profile page](https://demos.creative-tim.com/material-kit/examples/profile-page.html)
- [Landing page](https://demos.creative-tim.com/material-kit/examples/landing-page.html)

[View More](https://demos.creative-tim.com/material-kit/index.html).


## Quick start

- `npm i material-kit`
- [Download from Github](https://github.com/creativetimofficial/material-kit/archive/master.zip).
- [Download from Creative Tim](https://www.creative-tim.com/product/material-kit).
- Install with [Bower](https://bower.io/): ```bower install material-kit```.
- Clone the repo: `git clone https://github.com/creativetimofficial/material-kit.git`.


## Documentation
The documentation for the Material Kit is hosted at our [website](https://demos.creative-tim.com/material-kit/docs/2.1/getting-started/introduction.html).


## File Structure
Within the download you'll find the following directories and files:

```
material-kit/
├── CHANGELOG.md
├── LICENSE.md
├── README.md
├── assets
│   ├── assets-for-demo
│   ├── css
│   │   ├── material-kit.css
│   │   ├── material-kit.css.map
│   │   └── material-kit.min.css
│   ├── img
│   ├── js
│   │   ├── bootstrap-material-design.js
│   │   ├── bootstrap-material-design.min.js
│   │   ├── core
│   │   │   ├── jquery.min.js
│   │   │   └── popper.min.js
│   │   ├── material-kit.js
│   │   ├── material-kit.min.js
│   │   └── plugins
│   │       ├── bootstrap-datetimepicker.min.js
│   │       ├── jasny-bootstrap.min.js
│   │       ├── moment.min.js
│   │       └── nouislider.min.js
│   └── scss
│       ├── bootstrap
│       ├── core
│       ├── material-kit.scss
│       └── plugins
├── bower.json
├── docs
│   └── documentation.html
├── examples
│   ├── landing-page.html
│   ├── profile-page.html
│   └── signup-page.html
└── index.html
```


## Browser Support

At present, we officially aim to support the last two versions of the following browsers:

<img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/chrome.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/firefox.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/edge.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/safari.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/opera.png" width="64" height="64">



## Resources
- Demo: <https://demos.creative-tim.com/material-kit/index.html>
- Download Page: <https://www.creative-tim.com/product/material-kit>
- Documentation: <https://demos.creative-tim.com/material-kit/docs/2.1/getting-started/introduction.html>
- License Agreement: <https://www.creative-tim.com/license>
- Support: <https://www.creative-tim.com/contact-us>
- Issues: [Github Issues Page](https://github.com/creativetimofficial/material-kit/issues)
## Reporting Issues

We use GitHub Issues as the official bug tracker for the Material Dashboard. Here are some advices for our users that want to report an issue:

1. Make sure that you are using the latest version of the Material Dashboard. Check the CHANGELOG from your dashboard on our [website](https://www.creative-tim.com/).
2. Providing us reproducible steps for the issue will shorten the time it takes for it to be fixed.
3. Some issues may be browser specific, so specifying in what browser you encountered the issue might help.


## Technical Support or Questions

If you have questions or need help integrating the product please [contact us](https://www.creative-tim.com/contact-us) instead of opening an issue.



## Licensing

- Copyright 2018 Creative Tim (https://www.creative-tim.com/)

- Licensed under MIT (https://github.com/creativetimofficial/material-kit/blob/master/LICENSE.md)



## Useful Links

- [More products](https://www.creative-tim.com/bootstrap-themes) from Creative Tim
- [Tutorials](https://www.youtube.com/channel/UCVyTG4sCw-rOvB9oHkzZD1w)
- [Freebies](https://www.creative-tim.com/bootstrap-themes/free) from Creative Tim
- [Affiliate Program](https://www.creative-tim.com/affiliates/new) (earn money)

##### Social Media

Twitter: <https://twitter.com/CreativeTim>

Facebook: <https://www.facebook.com/CreativeTim>

Dribbble: <https://dribbble.com/creativetim>

Google+: <https://plus.google.com/+CreativetimPage>

Instagram: <https://www.instagram.com/CreativeTimOfficial>



CHANGELOG.md
# Change Log

## [2.0.5] 2018-11-23
### Improvements, BugFixing
- Changes for responsive devices
- Improvements and BugFixing for documentation
- Change versions on CSS and JS files
- Pro Documentation integrated in Free Version
- Eliminated extra CSS and JS

## [2.0.4] 2018-07-02
### BugFixing
- Solved responsive issues
- Updated plugins
- Speed optimizations
- Documentation errors fixed
- Gulp compiling of the SASS files
- Change versions on CSS and JS files
- Small changes for components

## [2.0.3] 2018-05-07
### BugFixing
- Scss cleaned and fixed issues when compiling
- Archive cleaned
- Other small bug fixing

## [2.0.2] 2018-02-13
### Improvements, Bootstrap 4.0.0 Final Included
- Reestablish the routes/folder for images

## [2.0.1] 2018-01-31
### Bug fixing, Improvements
- Added two plugins in Documentation
- Included gulp task that opens the project in browser
- Reestablish the routes for some Documentation Components
- responsive issue fixed for carousel

## [2.0.0] 2018-01-12
### Bootstrap 4 update
- Core base code on Bootstrap 4
- Documentation code base written from scratch
- Speed optimizations
- Class changes : .card-block to .card-body
- .navbar-toggleable-* to .navbar-expand-*
- .hidden-*-down to .d-none .d-*-block
- .hidden-*-up to .d-*-none
- .checkbox to .form-check
- .radio to .form-check
- more class changes here:https://medium.com/@lukaszholeczek/how-to-upgrade-bootstrap-4-alpha-6-to-bootstrap-4-beta-d43b4210f2a3
- Bug fixes for responsive devices
- Small changes for components

## [1.2.1] 2017-11-09
### BugFixing
- fixed parallax page scroll
- fixed cards from `ecommerce.html` page

## [1.2.0] 2017-09-13
### BugFixing
- jQuery library update to the latest version
- removed atv-img-animation.js library ( multiple problems with the browsers )
- fixed rotating card system for IE, changed rotating cards with raised image with regular cards with background image or plain cards
- bootstrap-selectpicker.js updated to the latest version

## [1.1.1] 2017-05-22
### BugFixing
- fixed problem with Morphing Cards(rotating manual and auto) and Perspective Cards when images were loading later than the javascript

## [1.1.0] 2017-04-14
### BugFixing, Plugins Updates, New Cards (HTML Version)
- v1.1.0 14 Apr, 2017 -
- Bootstrap v3.3.7
- Bootstrap DateTimePicker v4.17.45
- Replace Bootstrap SelectPicker v1.11.2
- Replace Tags with Bootstrap TagsInput v0.8.0
- jQuery.Flexisel v2.1.0
- NoUiSlider v9.0.0
- jQuery v3.1.1
- moment.js v2.14.1
- Added new cards
- Added 3D animation effect for images on cards(atvImg library)
- Added DynamicShadows
- Added New rotating cards
- Added Ecommerce Page
- Bug fixing
- Documentation Updates
- Added simple input example

IMPORTANT! If you migrate from V1.0.0 to V1.1.0 please make sure your replace the '.content' class inside each card with '.card-content', we've made this adjustment for a better control of the content that is inside a card. Sorry for the inconvenience.

## [1.0.0] 2016-06-15
### Initial Release


gulpfile.js
var gulp = require('gulp');
var path = require('path');
var sass = require('gulp-sass');
var autoprefixer = require('gulp-autoprefixer');
var sourcemaps = require('gulp-sourcemaps');
var open = require('gulp-open');

var Paths = {
  HERE: './',
  DIST: 'dist/',
  CSS: './assets/css/',
  SCSS_TOOLKIT_SOURCES: './assets/scss/material-kit.scss',
  SCSS: './assets/scss/**/**'
};

gulp.task('compile-scss', function() {
  return gulp.src(Paths.SCSS_TOOLKIT_SOURCES)
    .pipe(sourcemaps.init())
    .pipe(sass().on('error', sass.logError))
    .pipe(autoprefixer())
    .pipe(sourcemaps.write(Paths.HERE))
    .pipe(gulp.dest(Paths.CSS));
});

gulp.task('watch', function() {
  gulp.watch(Paths.SCSS, ['compile-scss']);
});

gulp.task('open', function() {
  gulp.src('index.html')
    .pipe(open());
});

gulp.task('open-app', ['open', 'watch']);

assets/js/mycustom.js
(() => {
  'use strict';
  // Page is loaded
  const objects = document.getElementsByClassName('asyncImage');

  Array.from(objects).map((item) => {
    // Start loading image
    const img = new Image();
    img.src = item.dataset.src;
    // Once image is loaded replace the src of the HTML element
    img.onload = () => {
      item.classList.remove('asyncImage');
      return item.nodeName === 'IMG' ?
        item.src = item.dataset.src :
        item.style.backgroundImage = `url(${item.dataset.src})`;
    };
  });
})();


/* card carousel -- specific navigation custom */
/* dibuat dengan cinta oleh apri */
/* dokumentasi : https://stackoverflow.com/questions/31224192/owl-carousel-making-custom-navigation */
$(document).ready(function () {
  $('.owl-carousel').each(function () {
    $(this).owlCarousel({
      loop: false,
      dots: false,
      responsiveClass: true,
      responsive: {
        // smartphone
        0: {
          items: 1,
          stagePadding: 20
        },
        370: {
          items: 1,
          margin: 10,
          stagePadding: 40
        },
        500: {
          items: 1,
          margin: -40,
          stagePadding: 40
        },
        // tablet
        600: {
          items: 2,
          margin: 150,
          stagePadding: 20
        },
        710: {
          items: 2,
          margin: 180,
          stagePadding: 40
        },
        // ipad pro
        1000: {
          items: 2,
          stagePadding: 80,
          nav: true,
          navText: [
            '<i class="material-icons" aria-hidden="true">keyboard_arrow_left</i><span class="sr-only">Previous</span>',
            '<i class="material-icons" aria-hidden="true">keyboard_arrow_right</i><span class="sr-only">Next</span>'
          ]
        },
        // laptop
        1300: {
          items: 3,
          stagePadding: 30,
          nav: true,
          navText: [
            '<i class="material-icons" aria-hidden="true">keyboard_arrow_left</i><span class="sr-only">Previous</span>',
            '<i class="material-icons" aria-hidden="true">keyboard_arrow_right</i><span class="sr-only">Next</span>'
          ]
        }
      }
    });
  });
});

assets/js/material-kit.js
/*!

 =========================================================
 * Material Kit - v2.0.5
 =========================================================

 * Product Page: https://www.creative-tim.com/product/material-kit
 * Copyright 2018 Creative Tim (http://www.creative-tim.com)

 * Designed by www.invisionapp.com Coded by www.creative-tim.com

 =========================================================

 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

 */

var big_image;

$(document).ready(function() {
  BrowserDetect.init();

  // Init Material scripts for buttons ripples, inputs animations etc, more info on the next link https://github.com/FezVrasta/bootstrap-material-design#materialjs
  $('body').bootstrapMaterialDesign();

  window_width = $(window).width();

  $navbar = $('.navbar[color-on-scroll]');
  scroll_distance = $navbar.attr('color-on-scroll') || 500;

  $navbar_collapse = $('.navbar').find('.navbar-collapse');

  //  Activate the Tooltips
  $('[data-toggle="tooltip"], [rel="tooltip"]').tooltip();

  // Activate Popovers
  $('[data-toggle="popover"]').popover();

  if ($('.navbar-color-on-scroll').length != 0) {
    $(window).on('scroll', materialKit.checkScrollForTransparentNavbar);
  }

  materialKit.checkScrollForTransparentNavbar();

  if (window_width >= 768) {
    big_image = $('.page-header[data-parallax="true"]');
    if (big_image.length != 0) {
      $(window).on('scroll', materialKit.checkScrollForParallax);
    }

  }


});

$(document).on('click', '.navbar-toggler', function() {
  $toggle = $(this);

  if (materialKit.misc.navbar_menu_visible == 1) {
    $('html').removeClass('nav-open');
    materialKit.misc.navbar_menu_visible = 0;
    $('#bodyClick').remove();
    setTimeout(function() {
      $toggle.removeClass('toggled');
    }, 550);

    $('html').removeClass('nav-open-absolute');
  } else {
    setTimeout(function() {
      $toggle.addClass('toggled');
    }, 580);


    div = '<div id="bodyClick"></div>';
    $(div).appendTo("body").click(function() {
      $('html').removeClass('nav-open');

      if ($('nav').hasClass('navbar-absolute')) {
        $('html').removeClass('nav-open-absolute');
      }
      materialKit.misc.navbar_menu_visible = 0;
      $('#bodyClick').remove();
      setTimeout(function() {
        $toggle.removeClass('toggled');
      }, 550);
    });

    if ($('nav').hasClass('navbar-absolute')) {
      $('html').addClass('nav-open-absolute');
    }

    $('html').addClass('nav-open');
    materialKit.misc.navbar_menu_visible = 1;
  }
});

materialKit = {
  misc: {
    navbar_menu_visible: 0,
    window_width: 0,
    transparent: true,
    fixedTop: false,
    navbar_initialized: false,
    isWindow: document.documentMode || /Edge/.test(navigator.userAgent)
  },

  initFormExtendedDatetimepickers: function() {
    $('.datetimepicker').datetimepicker({
      icons: {
        time: "fa fa-clock-o",
        date: "fa fa-calendar",
        up: "fa fa-chevron-up",
        down: "fa fa-chevron-down",
        previous: 'fa fa-chevron-left',
        next: 'fa fa-chevron-right',
        today: 'fa fa-screenshot',
        clear: 'fa fa-trash',
        close: 'fa fa-remove'
      }
    });
  },

  initSliders: function() {
    // Sliders for demo purpose
    var slider = document.getElementById('sliderRegular');

    noUiSlider.create(slider, {
      start: 40,
      connect: [true, false],
      range: {
        min: 0,
        max: 100
      }
    });

    var slider2 = document.getElementById('sliderDouble');

    noUiSlider.create(slider2, {
      start: [20, 60],
      connect: true,
      range: {
        min: 0,
        max: 100
      }
    });
  },

  checkScrollForParallax: function() {
    oVal = ($(window).scrollTop() / 3);
    big_image.css({
      'transform': 'translate3d(0,' + oVal + 'px,0)',
      '-webkit-transform': 'translate3d(0,' + oVal + 'px,0)',
      '-ms-transform': 'translate3d(0,' + oVal + 'px,0)',
      '-o-transform': 'translate3d(0,' + oVal + 'px,0)'
    });
  },

  checkScrollForTransparentNavbar: debounce(function() {
    if ($(document).scrollTop() > scroll_distance) {
      if (materialKit.misc.transparent) {
        materialKit.misc.transparent = false;
        $('.navbar-color-on-scroll').removeClass('navbar-transparent');
      }
    } else {
      if (!materialKit.misc.transparent) {
        materialKit.misc.transparent = true;
        $('.navbar-color-on-scroll').addClass('navbar-transparent');
      }
    }
  }, 17)
};

// Returns a function, that, as long as it continues to be invoked, will not
// be triggered. The function will be called after it stops being called for
// N milliseconds. If `immediate` is passed, trigger the function on the
// leading edge, instead of the trailing.

function debounce(func, wait, immediate) {
  var timeout;
  return function() {
    var context = this,
      args = arguments;
    clearTimeout(timeout);
    timeout = setTimeout(function() {
      timeout = null;
      if (!immediate) func.apply(context, args);
    }, wait);
    if (immediate && !timeout) func.apply(context, args);
  };
};

var BrowserDetect = {
  init: function() {
    this.browser = this.searchString(this.dataBrowser) || "Other";
    this.version = this.searchVersion(navigator.userAgent) || this.searchVersion(navigator.appVersion) || "Unknown";
  },
  searchString: function(data) {
    for (var i = 0; i < data.length; i++) {
      var dataString = data[i].string;
      this.versionSearchString = data[i].subString;

      if (dataString.indexOf(data[i].subString) !== -1) {
        return data[i].identity;
      }
    }
  },
  searchVersion: function(dataString) {
    var index = dataString.indexOf(this.versionSearchString);
    if (index === -1) {
      return;
    }

    var rv = dataString.indexOf("rv:");
    if (this.versionSearchString === "Trident" && rv !== -1) {
      return parseFloat(dataString.substring(rv + 3));
    } else {
      return parseFloat(dataString.substring(index + this.versionSearchString.length + 1));
    }
  },

  dataBrowser: [{
      string: navigator.userAgent,
      subString: "Chrome",
      identity: "Chrome"
    },
    {
      string: navigator.userAgent,
      subString: "MSIE",
      identity: "Explorer"
    },
    {
      string: navigator.userAgent,
      subString: "Trident",
      identity: "Explorer"
    },
    {
      string: navigator.userAgent,
      subString: "Firefox",
      identity: "Firefox"
    },
    {
      string: navigator.userAgent,
      subString: "Safari",
      identity: "Safari"
    },
    {
      string: navigator.userAgent,
      subString: "Opera",
      identity: "Opera"
    }
  ]

};

var better_browser = '<div class="container"><div class="better-browser row"><div class="col-md-2"></div><div class="col-md-8"><h3>We are sorry but it looks like your Browser doesn\'t support our website Features. In order to get the full experience please download a new version of your favourite browser.</h3></div><div class="col-md-2"></div><br><div class="col-md-4"><a href="https://www.mozilla.org/ro/firefox/new/" class="btn btn-warning">Mozilla</a><br></div><div class="col-md-4"><a href="https://www.google.com/chrome/browser/desktop/index.html" class="btn ">Chrome</a><br></div><div class="col-md-4"><a href="http://windows.microsoft.com/en-us/internet-explorer/ie-11-worldwide-languages" class="btn">Internet Explorer</a><br></div><br><br><h4>Thank you!</h4></div></div>';